packages/fx-core/tests/component/generator/generator.test.ts

```diff
@@ -541,15 +542,25 @@ describe("Generator happy path", async () => {
   });
 
   it("template from fallback", async () => {
-    const realTemplateName = "non-sso-tab";
+    const templateName = "test";
+    const mockFileName = "test.txt";
+    const mockFileData = "test data";
     const language = "ts";
+    const inputDir = path.join(tmpDir, "input");
+    const fallbackDir = path.join(tmpDir, "fallback");
+    await fs.ensureDir(path.join(inputDir, templateName));
+    await fs.ensureDir(fallbackDir);
+    await fs.writeFile(path.join(inputDir, templateName, mockFileName), mockFileData);
```

Comment:
Writing temp files to fs in UT is not a good practice. Be aware that you can always mock the fs operation to get rid of the dependency of disk. 
While I understand mocking fs operation takes much more effort, it's also acceptable to do so when you feel it is too tedious to do it perfectly.
In this case, I do not think it is necessary to write this file. Try using AdmZip.addFile.

Reply:
Thanks for the feedback. I'll try to minimize the fs code.